server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3083,21 +3084,44 @@ assert getActiveOperationsCount() == OPERATIONS_BLOCKED
         final TranslogStats translogStats = translogStats();
         // flush to make sure the latest commit, which will be opened by the read-only engine, includes all operations.
         flush(new FlushRequest().waitIfOngoing(true));
-        synchronized (mutex) {
-            verifyNotClosed();
-            // we must create a new engine under mutex (see IndexShard#snapshotStoreMetadata).
-            final Engine readOnlyEngine = new ReadOnlyEngine(newEngineConfig(), seqNoStats, translogStats, false, Function.identity());
-            IOUtils.close(currentEngineReference.getAndSet(readOnlyEngine));
-        }
 
         Engine newEngine = null;
+        SetOnce<Engine> newEngineReference = new SetOnce<>();
```

Comment:
Can we remove `newEngine` variable above?

Reply:
Fixed in 6888b6612c3f2b880afaffdd91f4865ec6433ab5